src/controller/p2p/preheat/controller.go

```diff
@@ -419,6 +421,21 @@ func (c *controller) DeletePolicy(ctx context.Context, id int64) error {
 	return c.pManager.Delete(ctx, id)
 }
 
+// DeletePoliciesOfProject deletes all the policy under project.
+func (c *controller) DeletePoliciesOfProject(ctx context.Context, project int64) error {
+	policies, err := c.ListPoliciesByProject(ctx, project, nil)
+	if err != nil {
+		return err
+	}
+
+	for _, p := range policies {
+		if err = c.DeletePolicy(ctx, p.ID); err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
How many SQLs will be executed here? Can we do this by one SQL?

Reply:
Good point. That may need to define new DAO and controller methods.

Reply:
@chlins 

What do you think?

Reply:
Yup, we can provide new DAO and manager method to do that.

Reply:
@heww @steven-zou After review the code, i think we will not have too many policies under the project, and in DeletePolicy, we do extra clean work like unschedule scheduler and delete executions for one policy. So keep the logic.

Reply:
ok